### ISSUE TITLE
Config file options are described in a different order than similar command line options

### ISSUE THREAD

**Issue by jolaf** [python/mypy#7575]
`mypy 0.740`

The order in which the command line options are documented at
https://mypy.readthedocs.io/en/latest/command_line.html
is different from the order the similarly named options are documented at
https://mypy.readthedocs.io/en/latest/config_file.html.
This inconsistency could be confusing.

In particular:

- `--warn-unused-configs` option is described in the beginning of the documentation in `Config file` section, but `warn_unused_configs` option is described  in the end of the documentation in `Miscellaneous`  section.
(WONTFIX)

- in `Import discovery` section the order of listing options is different.
(FIXED)

- `--no-site-packages` option is described in command line documentation, but is not mentioned in config file documentation and is not really working in config file.
(WONTFIX until  #7768 is resolved)

- In `Platform configuration` section the order of listing options is different.
(FIXED)

- `--warn-redundant-casts` option is listed under `Configuring warnings` section in command line documentation, but under `Miscellaneous` section in config file documentation.
(FIXED)

- `--allow-untyped-globals` options is not mentioned in config file documentation.
(FIXED)

- `--strict-optional-whitelist`, `--pretty`, `--sqlite-cache`, `--cache-fine-grained`, `--skip-cache-mtime-checks`, `--raise-exceptions`,  options are not mentioned in config file documentation.
(FIXED)

- `--lineprecision-report`, `--memory-xml-report`, `--xml-report` options are not mentioned in neither config file documentation nor command line documentation.
(FIXED, requires followup after #7775 is resolved)

- `--custom-typing` option is named `custom_typing_module` in config file documentation.
(WONTFIX until #7772 is resolved)

- `Report generation` section options are missing from config file documentation.
(FIXED)

- `follow_imports_for_stubs`, `show_none_errors`, `ignore_errors` options are not mentioned in neither command line documentation nor in `--help`.
(WONTFIX until #7774 is resolved)

- `no_implicit_optional` and `no_silence_site_packages` options should probably be documented without `no_`. Also, options without `no_` are not actually working, while they probably should.
(WONTFIX until #7773 is resolved)

- `new_semantic_analyzer` option should probably be removed from the config file documentation.
(FIXED)

**Comment by ilevkivskyi** [comment]
As in #7574: this doesn't look urgent, it would make sense to keep these more in sync.

**Comment by jolaf** [comment]
Now working on PR for this.
Would take some time as there's a lot of inconsistencies.

**Comment by jolaf** [comment]
See also https://github.com/python/mypy/issues/7768

**Comment by jolaf** [comment]
Also this suggesstion is implemented here:
https://github.com/python/mypy/issues/7574#issuecomment-536217974
